docs/navigation-advanced.md

```diff
@@ -0,0 +1,399 @@
+    }
+}
+```
+
+## NavigationBackStackEntry
+NavigationBackStackEntry represents an entry on the navigation backstack.
+You need to add this additional class.
+```csharp
+public class NavigationBackStackEntry
+{
+    public readonly string FrameKey;        
+
+    public readonly object Parameter;
+
+    public readonly Type SourcePageType;
+
+    public readonly NavigationTransitionInfo NavigationTransitionInfo;
+
+    public NavigationBackStackEntry(string frameKey, Type sourcePageType, NavigationConfig config)
+    {
+        FrameKey = frameKey;
+        SourcePageType = sourcePageType;
+        Parameter = config.Parameter;
+        NavigationTransitionInfo = config.InfoOverride;
+    }
+}
+```
+
+## NavigationConfig
+NavigationConfig represents the navigation configuration and allows you to specify navigation parameters and if you want to register the navigation on the back stack.
+
+You need to add this additional class.
+```csharp
+public class NavigationConfig
+{
+    public readonly bool RegisterOnBackStack;
+
+    public readonly object Parameter;
+
+    public readonly NavigationTransitionInfo InfoOverride;        
+
+    public NavigationConfig(bool registerOnBackStack = true, object parameter = null, NavigationTransitionInfo infoOverride = null)
+    {
+        RegisterOnBackStack = registerOnBackStack;
+        Parameter = parameter;
+        InfoOverride = infoOverride;            
+    }
+
+    public static NavigationConfig Default => new NavigationConfig();
+}
+```
+## Changes in project: 
+### Changes in ActivationService.cs
+- Initialize the main frame from `ActivateAsync()` replacing `Window.Current.Content = _shell?.Value ?? new Frame();` by 
+ ```csharp
+var frame = new Frame();
+if (_shell?.Value != null)
+{
+    frame.Content = _shell.Value;
+}
+NavigationService.InitializeMainFrame(frame);
+```
+
+- On the same method replace `NavigationService.Navigated += Frame_Navigated;` by `NavigationService.Navigated += OnNavigated;` and replace the event handler:
+
+ ```csharp
+ private void OnNavigated(object sender, NavigationArgs e)
+{
+    SystemNavigationManager.GetForCurrentView().AppViewBackButtonVisibility = NavigationService.CanGoBack ? AppViewBackButtonVisibility.Visible : AppViewBackButtonVisibility.Collapsed;
+}
+```
+### Changes in DefaultLaunchActivationHandler.cs
+-  Change HandleInternalAsync to 
+```csharp
+NavigationService.Navigate(_navElement, NavigationService.FrameKeyMain, new NavigationConfig(registerOnBackStack: true, parameter: args.Arguments));
+```
+
+- Change CanHandleInternal to 
+```csharp
+return !NavigationService.IsInitialized(NavigationService.FrameKeyMain);
+```
+
+## Using the advanced NavigationService
+
+### Navigate to startup page
+- On App.xaml.cs in CreateActivationService replace the existing code with `return new ActivationService(this, typeof(Views.StartUpPage));` to navigate directly to the StartupPage
+
+### Navigate from StartUpPage to ShellPage
```

Comment:
Where have you explained how StartUpPage and ShellPage are different and the purpose of each?